dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/ConstantPropagation.java

```diff
@@ -163,21 +171,19 @@ else if(current instanceof Load) {
     				e = EventFactory.newStore(newAddress, newValue, mo);
             	}
         	}
-// Locals are still causing problems
-        	
         	// Local events coming from assertions cause problems because the encoding of 
         	// AssertInline uses getResultRegisterExpr() which gets a value when calling
         	// Local.initialise() which is never the case for the new Event e below.
         	else if(current instanceof Local && ((Local)current).getExpr() instanceof IExpr && !current.is(Tag.ASSERTION)) {
-//        		Register reg = ((Local)current).getResultRegister();
-//        		
-//        		IExpr oldValue = (IExpr) ((Local)current).getExpr();
-//        		IExpr newValue = evaluate(oldValue, currentThreadId);
-//        		newValue = newValue instanceof ITop ? oldValue : newValue; 
-//        		Preconditions.checkState(newValue != null, 
-//        				String.format("Expression %s got no value after constant propagation analysis", oldValue));
-//        		
-//        		e = EventFactory.newLocal(reg, newValue);
+        		Register reg = ((Local)current).getResultRegister();
+        		
+        		IExpr oldValue = (IExpr) ((Local)current).getExpr();
+        		IExpr newValue = evaluate(oldValue, currentThreadId);
+        		newValue = newValue instanceof ITop ? oldValue : newValue; 
+        		Preconditions.checkState(newValue != null, 
+        				String.format("Expression %s got no value after constant propagation analysis", oldValue));
```

Comment:
For checking of invariants, use `Verify.verify` instead. Preconditions are meant to check for mistakes made by the caller, not by the algorithm itself.